// @adyen/api-library/lib/src/typings/index.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.terminal = exports.recurring = exports.platformsAccount = exports.platformsNotifications = exports.notification = exports.checkout = void 0;
/// <reference path="amount.ts" />
/// <reference path="apiError.ts" />
/// <reference path="applicationInfo.ts" />
/// <reference path="binLookup.ts" />
/// <reference path="enums/environment.ts" />
/// <reference path="enums/vatCategory.ts" />
/// <reference path="nexo.ts" />
/// <reference path="notification/notification.ts" />
/// <reference path="payouts.ts" />
/// <reference path="requestOptions.ts" />
/// <reference path="platformsFund.ts" />
/// <reference path="platformsHostedOnboardingPage.ts" />
/// <reference path="platformsNotificationConfiguration.ts" />
exports.checkout = __importStar(require("./checkout/models"));
exports.notification = __importStar(require("./notification/models"));
exports.platformsNotifications = __importStar(require("./platformsNotifications/models"));
exports.platformsAccount = __importStar(require("./platformsAccount/models"));
exports.recurring = __importStar(require("./recurring/models"));
exports.terminal = __importStar(require("./terminal/models"));
