// @adyen/api-library/lib/src/typings/checkout/accountInfo.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountInfo = void 0;
class AccountInfo {
}
exports.AccountInfo = AccountInfo;
(function (AccountInfo) {
    let AccountAgeIndicatorEnum;
    (function (AccountAgeIndicatorEnum) {
        AccountAgeIndicatorEnum[AccountAgeIndicatorEnum["NotApplicable"] = 'notApplicable'] = "NotApplicable";
        AccountAgeIndicatorEnum[AccountAgeIndicatorEnum["ThisTransaction"] = 'thisTransaction'] = "ThisTransaction";
        AccountAgeIndicatorEnum[AccountAgeIndicatorEnum["LessThan30Days"] = 'lessThan30Days'] = "LessThan30Days";
        AccountAgeIndicatorEnum[AccountAgeIndicatorEnum["From30To60Days"] = 'from30To60Days'] = "From30To60Days";
        AccountAgeIndicatorEnum[AccountAgeIndicatorEnum["MoreThan60Days"] = 'moreThan60Days'] = "MoreThan60Days";
    })(AccountAgeIndicatorEnum = AccountInfo.AccountAgeIndicatorEnum || (AccountInfo.AccountAgeIndicatorEnum = {}));
    let AccountChangeIndicatorEnum;
    (function (AccountChangeIndicatorEnum) {
        AccountChangeIndicatorEnum[AccountChangeIndicatorEnum["ThisTransaction"] = 'thisTransaction'] = "ThisTransaction";
        AccountChangeIndicatorEnum[AccountChangeIndicatorEnum["LessThan30Days"] = 'lessThan30Days'] = "LessThan30Days";
        AccountChangeIndicatorEnum[AccountChangeIndicatorEnum["From30To60Days"] = 'from30To60Days'] = "From30To60Days";
        AccountChangeIndicatorEnum[AccountChangeIndicatorEnum["MoreThan60Days"] = 'moreThan60Days'] = "MoreThan60Days";
    })(AccountChangeIndicatorEnum = AccountInfo.AccountChangeIndicatorEnum || (AccountInfo.AccountChangeIndicatorEnum = {}));
    let AccountTypeEnum;
    (function (AccountTypeEnum) {
        AccountTypeEnum[AccountTypeEnum["NotApplicable"] = 'notApplicable'] = "NotApplicable";
        AccountTypeEnum[AccountTypeEnum["Credit"] = 'credit'] = "Credit";
        AccountTypeEnum[AccountTypeEnum["Debit"] = 'debit'] = "Debit";
    })(AccountTypeEnum = AccountInfo.AccountTypeEnum || (AccountInfo.AccountTypeEnum = {}));
    let DeliveryAddressUsageIndicatorEnum;
    (function (DeliveryAddressUsageIndicatorEnum) {
        DeliveryAddressUsageIndicatorEnum[DeliveryAddressUsageIndicatorEnum["ThisTransaction"] = 'thisTransaction'] = "ThisTransaction";
        DeliveryAddressUsageIndicatorEnum[DeliveryAddressUsageIndicatorEnum["LessThan30Days"] = 'lessThan30Days'] = "LessThan30Days";
        DeliveryAddressUsageIndicatorEnum[DeliveryAddressUsageIndicatorEnum["From30To60Days"] = 'from30To60Days'] = "From30To60Days";
        DeliveryAddressUsageIndicatorEnum[DeliveryAddressUsageIndicatorEnum["MoreThan60Days"] = 'moreThan60Days'] = "MoreThan60Days";
    })(DeliveryAddressUsageIndicatorEnum = AccountInfo.DeliveryAddressUsageIndicatorEnum || (AccountInfo.DeliveryAddressUsageIndicatorEnum = {}));
    let PasswordChangeIndicatorEnum;
    (function (PasswordChangeIndicatorEnum) {
        PasswordChangeIndicatorEnum[PasswordChangeIndicatorEnum["NotApplicable"] = 'notApplicable'] = "NotApplicable";
        PasswordChangeIndicatorEnum[PasswordChangeIndicatorEnum["ThisTransaction"] = 'thisTransaction'] = "ThisTransaction";
        PasswordChangeIndicatorEnum[PasswordChangeIndicatorEnum["LessThan30Days"] = 'lessThan30Days'] = "LessThan30Days";
        PasswordChangeIndicatorEnum[PasswordChangeIndicatorEnum["From30To60Days"] = 'from30To60Days'] = "From30To60Days";
        PasswordChangeIndicatorEnum[PasswordChangeIndicatorEnum["MoreThan60Days"] = 'moreThan60Days'] = "MoreThan60Days";
    })(PasswordChangeIndicatorEnum = AccountInfo.PasswordChangeIndicatorEnum || (AccountInfo.PasswordChangeIndicatorEnum = {}));
    let PaymentAccountIndicatorEnum;
    (function (PaymentAccountIndicatorEnum) {
        PaymentAccountIndicatorEnum[PaymentAccountIndicatorEnum["NotApplicable"] = 'notApplicable'] = "NotApplicable";
        PaymentAccountIndicatorEnum[PaymentAccountIndicatorEnum["ThisTransaction"] = 'thisTransaction'] = "ThisTransaction";
        PaymentAccountIndicatorEnum[PaymentAccountIndicatorEnum["LessThan30Days"] = 'lessThan30Days'] = "LessThan30Days";
        PaymentAccountIndicatorEnum[PaymentAccountIndicatorEnum["From30To60Days"] = 'from30To60Days'] = "From30To60Days";
        PaymentAccountIndicatorEnum[PaymentAccountIndicatorEnum["MoreThan60Days"] = 'moreThan60Days'] = "MoreThan60Days";
    })(PaymentAccountIndicatorEnum = AccountInfo.PaymentAccountIndicatorEnum || (AccountInfo.PaymentAccountIndicatorEnum = {}));
})(AccountInfo = exports.AccountInfo || (exports.AccountInfo = {}));
