// @adyen/api-library/lib/src/typings/checkout/afterpayDetails.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AfterpayDetails = void 0;
class AfterpayDetails {
}
exports.AfterpayDetails = AfterpayDetails;
(function (AfterpayDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["AfterpayDefault"] = 'afterpay_default'] = "AfterpayDefault";
        TypeEnum[TypeEnum["Afterpaytouch"] = 'afterpaytouch'] = "Afterpaytouch";
        TypeEnum[TypeEnum["AfterpayB2b"] = 'afterpay_b2b'] = "AfterpayB2b";
    })(TypeEnum = AfterpayDetails.TypeEnum || (AfterpayDetails.TypeEnum = {}));
})(AfterpayDetails = exports.AfterpayDetails || (exports.AfterpayDetails = {}));
