// @adyen/api-library/lib/src/typings/checkout/avs.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Avs = void 0;
class Avs {
}
exports.Avs = Avs;
(function (Avs) {
    let EnabledEnum;
    (function (EnabledEnum) {
        EnabledEnum[EnabledEnum["Yes"] = 'yes'] = "Yes";
        EnabledEnum[EnabledEnum["No"] = 'no'] = "No";
        EnabledEnum[EnabledEnum["Automatic"] = 'automatic'] = "Automatic";
    })(EnabledEnum = Avs.EnabledEnum || (Avs.EnabledEnum = {}));
})(Avs = exports.Avs || (exports.Avs = {}));
