// @adyen/api-library/lib/src/typings/checkout/billDeskDetails.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillDeskDetails = void 0;
class BillDeskDetails {
}
exports.BillDeskDetails = BillDeskDetails;
(function (BillDeskDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["BilldeskOnline"] = 'billdesk_online'] = "BilldeskOnline";
        TypeEnum[TypeEnum["BilldeskWallet"] = 'billdesk_wallet'] = "BilldeskWallet";
        TypeEnum[TypeEnum["OnlinebankingIN"] = 'onlinebanking_IN'] = "OnlinebankingIN";
        TypeEnum[TypeEnum["WalletIN"] = 'wallet_IN'] = "WalletIN";
    })(TypeEnum = BillDeskDetails.TypeEnum || (BillDeskDetails.TypeEnum = {}));
})(BillDeskDetails = exports.BillDeskDetails || (exports.BillDeskDetails = {}));
