// @adyen/api-library/lib/src/typings/checkout/cardDetails.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardDetails = void 0;
class CardDetails {
}
exports.CardDetails = CardDetails;
(function (CardDetails) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum[FundingSourceEnum["Debit"] = 'debit'] = "Debit";
    })(FundingSourceEnum = CardDetails.FundingSourceEnum || (CardDetails.FundingSourceEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Scheme"] = 'scheme'] = "Scheme";
        TypeEnum[TypeEnum["NetworkToken"] = 'networkToken'] = "NetworkToken";
        TypeEnum[TypeEnum["Giftcard"] = 'giftcard'] = "Giftcard";
        TypeEnum[TypeEnum["Alliancedata"] = 'alliancedata'] = "Alliancedata";
        TypeEnum[TypeEnum["Card"] = 'card'] = "Card";
        TypeEnum[TypeEnum["Qiwiwallet"] = 'qiwiwallet'] = "Qiwiwallet";
        TypeEnum[TypeEnum["LianlianpayEbankingEnterprise"] = 'lianlianpay_ebanking_enterprise'] = "LianlianpayEbankingEnterprise";
        TypeEnum[TypeEnum["LianlianpayEbankingCredit"] = 'lianlianpay_ebanking_credit'] = "LianlianpayEbankingCredit";
        TypeEnum[TypeEnum["LianlianpayEbankingDebit"] = 'lianlianpay_ebanking_debit'] = "LianlianpayEbankingDebit";
        TypeEnum[TypeEnum["Entercash"] = 'entercash'] = "Entercash";
    })(TypeEnum = CardDetails.TypeEnum || (CardDetails.TypeEnum = {}));
})(CardDetails = exports.CardDetails || (exports.CardDetails = {}));
