// @adyen/api-library/lib/src/typings/checkout/deviceRenderOptions.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceRenderOptions = void 0;
class DeviceRenderOptions {
}
exports.DeviceRenderOptions = DeviceRenderOptions;
(function (DeviceRenderOptions) {
    let SdkInterfaceEnum;
    (function (SdkInterfaceEnum) {
        SdkInterfaceEnum[SdkInterfaceEnum["Native"] = 'native'] = "Native";
        SdkInterfaceEnum[SdkInterfaceEnum["Html"] = 'html'] = "Html";
        SdkInterfaceEnum[SdkInterfaceEnum["Both"] = 'both'] = "Both";
    })(SdkInterfaceEnum = DeviceRenderOptions.SdkInterfaceEnum || (DeviceRenderOptions.SdkInterfaceEnum = {}));
    let SdkUiTypeEnum;
    (function (SdkUiTypeEnum) {
        SdkUiTypeEnum[SdkUiTypeEnum["MultiSelect"] = 'multiSelect'] = "MultiSelect";
        SdkUiTypeEnum[SdkUiTypeEnum["OtherHtml"] = 'otherHtml'] = "OtherHtml";
        SdkUiTypeEnum[SdkUiTypeEnum["OutOfBand"] = 'outOfBand'] = "OutOfBand";
        SdkUiTypeEnum[SdkUiTypeEnum["SingleSelect"] = 'singleSelect'] = "SingleSelect";
        SdkUiTypeEnum[SdkUiTypeEnum["Text"] = 'text'] = "Text";
    })(SdkUiTypeEnum = DeviceRenderOptions.SdkUiTypeEnum || (DeviceRenderOptions.SdkUiTypeEnum = {}));
})(DeviceRenderOptions = exports.DeviceRenderOptions || (exports.DeviceRenderOptions = {}));
