// @adyen/api-library/lib/src/typings/checkout/dokuDetails.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DokuDetails = void 0;
class DokuDetails {
}
exports.DokuDetails = DokuDetails;
(function (DokuDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["MandiriVa"] = 'doku_mandiri_va'] = "MandiriVa";
        TypeEnum[TypeEnum["CimbVa"] = 'doku_cimb_va'] = "CimbVa";
        TypeEnum[TypeEnum["DanamonVa"] = 'doku_danamon_va'] = "DanamonVa";
        TypeEnum[TypeEnum["BniVa"] = 'doku_bni_va'] = "BniVa";
        TypeEnum[TypeEnum["PermataLiteAtm"] = 'doku_permata_lite_atm'] = "PermataLiteAtm";
        TypeEnum[TypeEnum["BriVa"] = 'doku_bri_va'] = "BriVa";
        TypeEnum[TypeEnum["BcaVa"] = 'doku_bca_va'] = "BcaVa";
        TypeEnum[TypeEnum["Alfamart"] = 'doku_alfamart'] = "Alfamart";
        TypeEnum[TypeEnum["Indomaret"] = 'doku_indomaret'] = "Indomaret";
    })(TypeEnum = DokuDetails.TypeEnum || (DokuDetails.TypeEnum = {}));
})(DokuDetails = exports.DokuDetails || (exports.DokuDetails = {}));
