// @adyen/api-library/lib/src/typings/checkout/dragonpayDetails.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DragonpayDetails = void 0;
class DragonpayDetails {
}
exports.DragonpayDetails = DragonpayDetails;
(function (DragonpayDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Ebanking"] = 'dragonpay_ebanking'] = "Ebanking";
        TypeEnum[TypeEnum["OtcBanking"] = 'dragonpay_otc_banking'] = "OtcBanking";
        TypeEnum[TypeEnum["OtcNonBanking"] = 'dragonpay_otc_non_banking'] = "OtcNonBanking";
        TypeEnum[TypeEnum["OtcPhilippines"] = 'dragonpay_otc_philippines'] = "OtcPhilippines";
    })(TypeEnum = DragonpayDetails.TypeEnum || (DragonpayDetails.TypeEnum = {}));
})(DragonpayDetails = exports.DragonpayDetails || (exports.DragonpayDetails = {}));
