// @adyen/api-library/lib/src/typings/checkout/genericIssuerPaymentMethodDetails.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenericIssuerPaymentMethodDetails = void 0;
class GenericIssuerPaymentMethodDetails {
}
exports.GenericIssuerPaymentMethodDetails = GenericIssuerPaymentMethodDetails;
(function (GenericIssuerPaymentMethodDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Eps"] = 'eps'] = "Eps";
        TypeEnum[TypeEnum["OnlineBankingSK"] = 'onlineBanking_SK'] = "OnlineBankingSK";
        TypeEnum[TypeEnum["OnlineBankingCZ"] = 'onlineBanking_CZ'] = "OnlineBankingCZ";
    })(TypeEnum = GenericIssuerPaymentMethodDetails.TypeEnum || (GenericIssuerPaymentMethodDetails.TypeEnum = {}));
})(GenericIssuerPaymentMethodDetails = exports.GenericIssuerPaymentMethodDetails || (exports.GenericIssuerPaymentMethodDetails = {}));
