// @adyen/api-library/lib/src/typings/checkout/klarnaDetails.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KlarnaDetails = void 0;
class KlarnaDetails {
}
exports.KlarnaDetails = KlarnaDetails;
(function (KlarnaDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Klarna"] = 'klarna'] = "Klarna";
        TypeEnum[TypeEnum["Klarnapayments"] = 'klarnapayments'] = "Klarnapayments";
        TypeEnum[TypeEnum["KlarnapaymentsAccount"] = 'klarnapayments_account'] = "KlarnapaymentsAccount";
        TypeEnum[TypeEnum["KlarnapaymentsB2b"] = 'klarnapayments_b2b'] = "KlarnapaymentsB2b";
        TypeEnum[TypeEnum["KlarnaPaynow"] = 'klarna_paynow'] = "KlarnaPaynow";
        TypeEnum[TypeEnum["KlarnaAccount"] = 'klarna_account'] = "KlarnaAccount";
        TypeEnum[TypeEnum["KlarnaB2b"] = 'klarna_b2b'] = "KlarnaB2b";
    })(TypeEnum = KlarnaDetails.TypeEnum || (KlarnaDetails.TypeEnum = {}));
})(KlarnaDetails = exports.KlarnaDetails || (exports.KlarnaDetails = {}));
