// @adyen/api-library/lib/src/typings/checkout/mandate.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Mandate = void 0;
class Mandate {
}
exports.Mandate = Mandate;
(function (Mandate) {
    let AmountRuleEnum;
    (function (AmountRuleEnum) {
        AmountRuleEnum[AmountRuleEnum["Max"] = 'max'] = "Max";
        AmountRuleEnum[AmountRuleEnum["Exact"] = 'exact'] = "Exact";
    })(AmountRuleEnum = Mandate.AmountRuleEnum || (Mandate.AmountRuleEnum = {}));
    let BillingAttemptsRuleEnum;
    (function (BillingAttemptsRuleEnum) {
        BillingAttemptsRuleEnum[BillingAttemptsRuleEnum["On"] = 'on'] = "On";
        BillingAttemptsRuleEnum[BillingAttemptsRuleEnum["Before"] = 'before'] = "Before";
        BillingAttemptsRuleEnum[BillingAttemptsRuleEnum["After"] = 'after'] = "After";
    })(BillingAttemptsRuleEnum = Mandate.BillingAttemptsRuleEnum || (Mandate.BillingAttemptsRuleEnum = {}));
    let FrequencyEnum;
    (function (FrequencyEnum) {
        FrequencyEnum[FrequencyEnum["Adhoc"] = 'adhoc'] = "Adhoc";
        FrequencyEnum[FrequencyEnum["Daily"] = 'daily'] = "Daily";
        FrequencyEnum[FrequencyEnum["Weekly"] = 'weekly'] = "Weekly";
        FrequencyEnum[FrequencyEnum["BiWeekly"] = 'biWeekly'] = "BiWeekly";
        FrequencyEnum[FrequencyEnum["Monthly"] = 'monthly'] = "Monthly";
        FrequencyEnum[FrequencyEnum["Quarterly"] = 'quarterly'] = "Quarterly";
        FrequencyEnum[FrequencyEnum["HalfYearly"] = 'halfYearly'] = "HalfYearly";
        FrequencyEnum[FrequencyEnum["Yearly"] = 'yearly'] = "Yearly";
    })(FrequencyEnum = Mandate.FrequencyEnum || (Mandate.FrequencyEnum = {}));
})(Mandate = exports.Mandate || (exports.Mandate = {}));
