// @adyen/api-library/lib/src/typings/checkout/openInvoiceDetails.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenInvoiceDetails = void 0;
class OpenInvoiceDetails {
}
exports.OpenInvoiceDetails = OpenInvoiceDetails;
(function (OpenInvoiceDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Openinvoice"] = 'openinvoice'] = "Openinvoice";
        TypeEnum[TypeEnum["AfterpayDirectdebit"] = 'afterpay_directdebit'] = "AfterpayDirectdebit";
        TypeEnum[TypeEnum["AtomePos"] = 'atome_pos'] = "AtomePos";
    })(TypeEnum = OpenInvoiceDetails.TypeEnum || (OpenInvoiceDetails.TypeEnum = {}));
})(OpenInvoiceDetails = exports.OpenInvoiceDetails || (exports.OpenInvoiceDetails = {}));
