// @adyen/api-library/lib/src/typings/checkout/payPalDetails.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayPalDetails = void 0;
class PayPalDetails {
}
exports.PayPalDetails = PayPalDetails;
(function (PayPalDetails) {
    let SubtypeEnum;
    (function (SubtypeEnum) {
        SubtypeEnum[SubtypeEnum["Redirect"] = 'redirect'] = "Redirect";
        SubtypeEnum[SubtypeEnum["Sdk"] = 'sdk'] = "Sdk";
    })(SubtypeEnum = PayPalDetails.SubtypeEnum || (PayPalDetails.SubtypeEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Paypal"] = 'paypal'] = "Paypal";
    })(TypeEnum = PayPalDetails.TypeEnum || (PayPalDetails.TypeEnum = {}));
})(PayPalDetails = exports.PayPalDetails || (exports.PayPalDetails = {}));
