// @adyen/api-library/lib/src/typings/checkout/recurring.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Recurring = void 0;
class Recurring {
}
exports.Recurring = Recurring;
(function (Recurring) {
    let ContractEnum;
    (function (ContractEnum) {
        ContractEnum[ContractEnum["ONECLICK"] = 'ONECLICK'] = "ONECLICK";
        ContractEnum[ContractEnum["RECURRING"] = 'RECURRING'] = "RECURRING";
        ContractEnum[ContractEnum["PAYOUT"] = 'PAYOUT'] = "PAYOUT";
    })(ContractEnum = Recurring.ContractEnum || (Recurring.ContractEnum = {}));
    let TokenServiceEnum;
    (function (TokenServiceEnum) {
        TokenServiceEnum[TokenServiceEnum["VISATOKENSERVICE"] = 'VISATOKENSERVICE'] = "VISATOKENSERVICE";
        TokenServiceEnum[TokenServiceEnum["MCTOKENSERVICE"] = 'MCTOKENSERVICE'] = "MCTOKENSERVICE";
    })(TokenServiceEnum = Recurring.TokenServiceEnum || (Recurring.TokenServiceEnum = {}));
})(Recurring = exports.Recurring || (exports.Recurring = {}));
