// @adyen/api-library/lib/src/typings/checkout/shopperInput.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShopperInput = void 0;
class ShopperInput {
}
exports.ShopperInput = ShopperInput;
(function (ShopperInput) {
    let BillingAddressEnum;
    (function (BillingAddressEnum) {
        BillingAddressEnum[BillingAddressEnum["Editable"] = 'editable'] = "Editable";
        BillingAddressEnum[BillingAddressEnum["Hidden"] = 'hidden'] = "Hidden";
        BillingAddressEnum[BillingAddressEnum["ReadOnly"] = 'readOnly'] = "ReadOnly";
    })(BillingAddressEnum = ShopperInput.BillingAddressEnum || (ShopperInput.BillingAddressEnum = {}));
    let DeliveryAddressEnum;
    (function (DeliveryAddressEnum) {
        DeliveryAddressEnum[DeliveryAddressEnum["Editable"] = 'editable'] = "Editable";
        DeliveryAddressEnum[DeliveryAddressEnum["Hidden"] = 'hidden'] = "Hidden";
        DeliveryAddressEnum[DeliveryAddressEnum["ReadOnly"] = 'readOnly'] = "ReadOnly";
    })(DeliveryAddressEnum = ShopperInput.DeliveryAddressEnum || (ShopperInput.DeliveryAddressEnum = {}));
    let PersonalDetailsEnum;
    (function (PersonalDetailsEnum) {
        PersonalDetailsEnum[PersonalDetailsEnum["Editable"] = 'editable'] = "Editable";
        PersonalDetailsEnum[PersonalDetailsEnum["Hidden"] = 'hidden'] = "Hidden";
        PersonalDetailsEnum[PersonalDetailsEnum["ReadOnly"] = 'readOnly'] = "ReadOnly";
    })(PersonalDetailsEnum = ShopperInput.PersonalDetailsEnum || (ShopperInput.PersonalDetailsEnum = {}));
})(ShopperInput = exports.ShopperInput || (exports.ShopperInput = {}));
