// @adyen/api-library/lib/src/typings/checkout/storedPaymentMethodDetails.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredPaymentMethodDetails = void 0;
class StoredPaymentMethodDetails {
}
exports.StoredPaymentMethodDetails = StoredPaymentMethodDetails;
(function (StoredPaymentMethodDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["BcmcMobile"] = 'bcmc_mobile'] = "BcmcMobile";
        TypeEnum[TypeEnum["BcmcMobileQR"] = 'bcmc_mobile_QR'] = "BcmcMobileQR";
        TypeEnum[TypeEnum["BcmcMobileApp"] = 'bcmc_mobile_app'] = "BcmcMobileApp";
        TypeEnum[TypeEnum["MomoWallet"] = 'momo_wallet'] = "MomoWallet";
        TypeEnum[TypeEnum["MomoWalletApp"] = 'momo_wallet_app'] = "MomoWalletApp";
        TypeEnum[TypeEnum["PaymayaWallet"] = 'paymaya_wallet'] = "PaymayaWallet";
        TypeEnum[TypeEnum["GrabpaySG"] = 'grabpay_SG'] = "GrabpaySG";
        TypeEnum[TypeEnum["GrabpayMY"] = 'grabpay_MY'] = "GrabpayMY";
        TypeEnum[TypeEnum["GrabpayTH"] = 'grabpay_TH'] = "GrabpayTH";
        TypeEnum[TypeEnum["GrabpayID"] = 'grabpay_ID'] = "GrabpayID";
        TypeEnum[TypeEnum["GrabpayVN"] = 'grabpay_VN'] = "GrabpayVN";
        TypeEnum[TypeEnum["GrabpayPH"] = 'grabpay_PH'] = "GrabpayPH";
        TypeEnum[TypeEnum["Oxxo"] = 'oxxo'] = "Oxxo";
        TypeEnum[TypeEnum["Gcash"] = 'gcash'] = "Gcash";
        TypeEnum[TypeEnum["Kakaopay"] = 'kakaopay'] = "Kakaopay";
        TypeEnum[TypeEnum["Truemoney"] = 'truemoney'] = "Truemoney";
    })(TypeEnum = StoredPaymentMethodDetails.TypeEnum || (StoredPaymentMethodDetails.TypeEnum = {}));
})(StoredPaymentMethodDetails = exports.StoredPaymentMethodDetails || (exports.StoredPaymentMethodDetails = {}));
