// @adyen/api-library/lib/src/typings/checkout/threeDSecureData.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreeDSecureData = void 0;
class ThreeDSecureData {
}
exports.ThreeDSecureData = ThreeDSecureData;
(function (ThreeDSecureData) {
    let AuthenticationResponseEnum;
    (function (AuthenticationResponseEnum) {
        AuthenticationResponseEnum[AuthenticationResponseEnum["Y"] = 'Y'] = "Y";
        AuthenticationResponseEnum[AuthenticationResponseEnum["N"] = 'N'] = "N";
        AuthenticationResponseEnum[AuthenticationResponseEnum["U"] = 'U'] = "U";
        AuthenticationResponseEnum[AuthenticationResponseEnum["A"] = 'A'] = "A";
    })(AuthenticationResponseEnum = ThreeDSecureData.AuthenticationResponseEnum || (ThreeDSecureData.AuthenticationResponseEnum = {}));
    let ChallengeCancelEnum;
    (function (ChallengeCancelEnum) {
        ChallengeCancelEnum[ChallengeCancelEnum["_01"] = '01'] = "_01";
        ChallengeCancelEnum[ChallengeCancelEnum["_02"] = '02'] = "_02";
        ChallengeCancelEnum[ChallengeCancelEnum["_03"] = '03'] = "_03";
        ChallengeCancelEnum[ChallengeCancelEnum["_04"] = '04'] = "_04";
        ChallengeCancelEnum[ChallengeCancelEnum["_05"] = '05'] = "_05";
        ChallengeCancelEnum[ChallengeCancelEnum["_06"] = '06'] = "_06";
        ChallengeCancelEnum[ChallengeCancelEnum["_07"] = '07'] = "_07";
    })(ChallengeCancelEnum = ThreeDSecureData.ChallengeCancelEnum || (ThreeDSecureData.ChallengeCancelEnum = {}));
    let DirectoryResponseEnum;
    (function (DirectoryResponseEnum) {
        DirectoryResponseEnum[DirectoryResponseEnum["A"] = 'A'] = "A";
        DirectoryResponseEnum[DirectoryResponseEnum["C"] = 'C'] = "C";
        DirectoryResponseEnum[DirectoryResponseEnum["D"] = 'D'] = "D";
        DirectoryResponseEnum[DirectoryResponseEnum["I"] = 'I'] = "I";
        DirectoryResponseEnum[DirectoryResponseEnum["N"] = 'N'] = "N";
        DirectoryResponseEnum[DirectoryResponseEnum["R"] = 'R'] = "R";
        DirectoryResponseEnum[DirectoryResponseEnum["U"] = 'U'] = "U";
        DirectoryResponseEnum[DirectoryResponseEnum["Y"] = 'Y'] = "Y";
    })(DirectoryResponseEnum = ThreeDSecureData.DirectoryResponseEnum || (ThreeDSecureData.DirectoryResponseEnum = {}));
})(ThreeDSecureData = exports.ThreeDSecureData || (exports.ThreeDSecureData = {}));
