// @adyen/api-library/lib/src/typings/checkout/threeDS2Result.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2022 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen Checkout API
 *
 * The version of the OpenAPI document: v69
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreeDS2Result = void 0;
class ThreeDS2Result {
}
exports.ThreeDS2Result = ThreeDS2Result;
(function (ThreeDS2Result) {
    let ChallengeCancelEnum;
    (function (ChallengeCancelEnum) {
        ChallengeCancelEnum[ChallengeCancelEnum["_01"] = '01'] = "_01";
        ChallengeCancelEnum[ChallengeCancelEnum["_02"] = '02'] = "_02";
        ChallengeCancelEnum[ChallengeCancelEnum["_03"] = '03'] = "_03";
        ChallengeCancelEnum[ChallengeCancelEnum["_04"] = '04'] = "_04";
        ChallengeCancelEnum[ChallengeCancelEnum["_05"] = '05'] = "_05";
        ChallengeCancelEnum[ChallengeCancelEnum["_06"] = '06'] = "_06";
        ChallengeCancelEnum[ChallengeCancelEnum["_07"] = '07'] = "_07";
    })(ChallengeCancelEnum = ThreeDS2Result.ChallengeCancelEnum || (ThreeDS2Result.ChallengeCancelEnum = {}));
    let ChallengeIndicatorEnum;
    (function (ChallengeIndicatorEnum) {
        ChallengeIndicatorEnum[ChallengeIndicatorEnum["NoPreference"] = 'noPreference'] = "NoPreference";
        ChallengeIndicatorEnum[ChallengeIndicatorEnum["RequestNoChallenge"] = 'requestNoChallenge'] = "RequestNoChallenge";
        ChallengeIndicatorEnum[ChallengeIndicatorEnum["RequestChallenge"] = 'requestChallenge'] = "RequestChallenge";
        ChallengeIndicatorEnum[ChallengeIndicatorEnum["RequestChallengeAsMandate"] = 'requestChallengeAsMandate'] = "RequestChallengeAsMandate";
    })(ChallengeIndicatorEnum = ThreeDS2Result.ChallengeIndicatorEnum || (ThreeDS2Result.ChallengeIndicatorEnum = {}));
    let ExemptionIndicatorEnum;
    (function (ExemptionIndicatorEnum) {
        ExemptionIndicatorEnum[ExemptionIndicatorEnum["LowValue"] = 'lowValue'] = "LowValue";
        ExemptionIndicatorEnum[ExemptionIndicatorEnum["SecureCorporate"] = 'secureCorporate'] = "SecureCorporate";
        ExemptionIndicatorEnum[ExemptionIndicatorEnum["TrustedBeneficiary"] = 'trustedBeneficiary'] = "TrustedBeneficiary";
        ExemptionIndicatorEnum[ExemptionIndicatorEnum["TransactionRiskAnalysis"] = 'transactionRiskAnalysis'] = "TransactionRiskAnalysis";
    })(ExemptionIndicatorEnum = ThreeDS2Result.ExemptionIndicatorEnum || (ThreeDS2Result.ExemptionIndicatorEnum = {}));
})(ThreeDS2Result = exports.ThreeDS2Result || (exports.ThreeDS2Result = {}));
